### PR TITLE
Migrate to RESTEasy Reactive

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
This port is causing two test to fail in the backend module:

```
[ERROR] Failures: 
[ERROR]   AlertingServiceTest.testLogging:127 expected: <true> but was: <false>
[ERROR]   TokenAuthTest.testTokenInHeader:42 1 expectation failed.
Expected status code <200> but was <401>.
```

I have no idea about the first one, but I believe the second one is a bad test and should be failing

File: pom.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
         <quinoa.version>1.0.5</quinoa.version>
         <surefire-plugin.version>3.0.0-M6</surefire-plugin.version>
         <failsafe-plugin.version>3.0.0-M5</failsafe-plugin.version>
-        <keycloak-admin-client.version>18.0.0</keycloak-admin-client.version>
```
Comment (rvansa) [inline pom.xml:85]: Why did you downgrade Keycloak client? I think I've updated this due to some vulnerabilities detected by Snyk. The newer client should be still compatible with older Keycloak.

Comment (geoand) [reply pom.xml:85]: The reason was that the version needs to align to what Quarkus uses, otherwise there is an enforcer rule that fails

**Comment by rvansa** [comment]
@geoand Thanks for this PR - and sorry for some delay in review due to holidays. I think that the devil will be in ordering the filters before whatever handles OIDC.

**Comment by geoand** [comment]
Hey,

I am just coming back from a sick leave, so it might take me a while to get back to this :)

File: pom.xml (hunk 1 above)
Comment (rvansa) [reply pom.xml:85]: Okay, in that case we could exclude the dependency brought in Quarkus and use the newer one.

File: horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public void testTokenInHeader() {
       given().get("/api/test/" + test.id + "?token=blabla").then().statusCode(404);
 
       given().header(TokenInterceptor.TOKEN_HEADER, "foobar").get("/api/test/" + test.id).then().statusCode(200);
-      given().header(HttpHeaders.AUTHORIZATION, "Bearer foobar").get("/api/test/" + test.id).then().statusCode(200);
```
[the remark this answers is not in the document]
Comment (geoand) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: I'll have to take a closer look at this

File: pom.xml (hunk 1 above)
Comment (geoand) [reply pom.xml:85]: That should be fine as well

File: horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (geoand) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: >  this is not supposed to be handled by OIDC server. HorreumAuthorizationFilter should run before the OIDC authorization kicks in

How is this ensured?

[the remark this answers is not in the document]
Comment (rvansa) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: I can't exactly say - I make sure that `HorreumAuthorizationFilter` is the first of filters through `META-INF/web.xml` (it is not possible to do that just through annotations). I don't remember where OIDC is in the stack (I was debugging to see that before) but it happens in this order.

[the remark this answers is not in the document]
Comment (geoand) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: Understood. It probably happens by accident. I'll have to take another look.

[the remark this answers is not in the document]
Comment (rvansa) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: So how is OIDC integrated into the reactive stack? Can you order something explicitly before that?

[the remark this answers is not in the document]
Comment (geoand) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/TokenAuthTest.java:42]: It's a Vert.x handler that kicks in quite early.

**Comment by rvansa** [comment]
Hi @geoand, would you like to finish the request or should I take it over?

**Comment by geoand** [comment]
Hi,

I'm on PTO and a lot of things have pilled up, so realistically I won't be
able to get to this for another couple of weeks.

So up to you 😎

**Comment by rvansa** [comment]
So, I've managed to get the test working with few changes in `HorreumAuthorizationFilter`: first I set it up with
```
@ServerRequestFilter(priority = Priorities.AUTHENTICATION - 1)
```
and then I remove the Authorization header with this:
```
         try {
            Field delegate = PreventAbortResteasyReactiveContainerRequestContext.class.getDeclaredField("delegate");
            delegate.setAccessible(true);
            var actualContext = (ResteasyReactiveContainerRequestContext) delegate.get(containerRequestContext);
            HttpServerRequest actualServerRequest = ((QuarkusResteasyReactiveRequestContext) actualContext.getServerRequestContext()).vertxServerRequest();
            actualServerRequest.headers().remove(HttpHeaders.AUTHORIZATION);
         } catch (NoSuchFieldException | IllegalAccessException e) {
            e.printStackTrace();  // TODO: Customise this generated block
         }
```
I wouldn't mind the few casts, but so far I haven't found a way around accessing the delegate...

**Comment by geoand** [comment]
We'll, we can make the method parameter "injection" work with whatever type we want in RESTEasy Reactive 😉

**Comment by geoand** [comment]
🎉

**Comment by rvansa** [comment]
@geoand I forgot to thank you for this, you did most of the heavy lifting.

**Comment by geoand** [comment]
Thank you for taking it over the finish line!

Glad to see this in :)

**Pull request merged**